scraper.rb

```diff
@@ -1,138 +1,240 @@
-
-  # build the sentence with new sign up stats
-  text = new_signups_last_fortnight.to_s +
-        " people signed up for PlanningAlerts last fortnight :revolving_hearts:"
-  text += " " + change_sentence(new_signups_last_fortnight, new_signups_fortnight_before_last) + "\n"
-  text += unsubscribers_last_fortnight.to_s + " people left. "
-  text += change_sentence(unsubscribers_last_fortnight, unsubscribers_fortnight_before_last) + "\n"
-  text += "You shipped #{commits_count} commits in the same period.\n" unless commits_count.zero?
-  text += "There are now " + ActiveSupport::NumberHelper.number_to_human(total_planningalerts_subscribers).downcase +
-          " PlanningAlerts subscribers! :star2:"
-
-  puts "Post the message to Slack"
-  if post_message_to_slack(text) === "ok"
-    # record the message and the date sent to the db
-    puts "Save the message to the db"
-
-    if ENV["MORPH_LIVE_MODE"].eql? "true"
-      ScraperWiki.save_sqlite([:date_posted], {date_posted: Date.today.to_s, text: text})
-    else
-      puts text
+    def scrape_and_post_message
+      message = build_message
+
+      if morph_live_mode?
+        puts 'Posting the message to Slack'
+        post(message)
+      else
+        puts 'Not posting to Slack'
+        puts 'Not recording the message in the database'
+        print(message)
+      end
+    end
+
+    def post_message_to_slack(text, opts = {})
+      options = {
+        username: 'Jacaranda',
+        text: text
+      }.merge(opts)
+      url = options.delete(:url)
+      raise ArgumentError, 'Must supply :url in options' unless url
+
+      RestClient.post(url, options.to_json) =~ /ok/i
+    end
+
+    def last_fortnight
+      start  = 1.fortnight.ago.beginning_of_week.to_date
+      finish = 1.week.ago.end_of_week.to_date
+      (start..finish).to_a
+    end
+
+    def build_message
+      [
+        PlanningAlerts.new_subscribers_text(period: last_fortnight),
+        PlanningAlerts.new_unsubscribers_text(period: last_fortnight),
+        GitHub.commits_text(period: last_fortnight),
+        PlanningAlerts.total_subscribers_text
+      ].compact.join("\n\n")
+    end
+
+    def post(message)
+      opts = { url: ENV['MORPH_SLACK_CHANNEL_WEBHOOK_URL'] }
+      opts[:channel] = '#bottesting' unless morph_live_mode?
+      if post_message_to_slack(message, opts)
+        puts 'Recording the message in the database'
+        record_successful_post(message)
+      else
+        puts 'Error: could not post the message to Slack!'
+      end
+    end
+
+    def record_successful_post(message)
+      ScraperWiki.save_sqlite([:date_posted], date_posted: Date.today.to_s, text: message)
+    end
+
+    def print(message)
+      puts
+      puts message.gsub(/^/m, '> ')
     end
   end
-else
-  p "I’ve already spoken to the team this fortnight"
 end
+
+Scraper.run if $PROGRAM_NAME == __FILE__
```

Comment:
If the program is being run directly `ruby scraper.rb` it will run the scraper. 

If the program is required (like it is from the test suite), don't run the scraper.